src/backend/compiler/compiler.cc

```diff
@@ -990,38 +991,69 @@ extern "C" void seashell_preprocessor_free(struct seashell_preprocessor *preproc
 }
 
 /**
- * seashell_preprocessor_set_main_file(struct seashell_preprocessor* preprocessor, const char* file)
- * Sets the main file to begin running the preprocessor on
+ * seashell_preprocessor_set_question_dir(struct seashell_preprocessor* preprocessor, const char *dir)
```

Comment:
Yeah I remember you mentioning that, but in the case where the header file is in a different directory from its implementation we need to know the question dir anyway (and it can't be retrieved from the #include). And I think it's possible that some assignment question in Seashell might be set up in this way in the future. I can clean up the code a bit by passing the paths and re-using them wherever they're needed.